clash-lib/src/Clash/Primitives/DSL.hs

```diff
@@ -268,28 +268,32 @@ boolFromBitVector
   -> Identifier
   -> TExpr
   -> State (BlockState backend) TExpr
-boolFromBitVector n = outputCoerce (BitVector n) Bool (\i -> "unsigned(" <> i <> ") > 0")
+boolFromBitVector n =
+  outputCoerce (BitVector n) Bool (\i -> "unsigned(" <> i <> ") > 0")
 
 -- | Used to create an output `Unsigned` from a `BitVector` of given
 -- size. Works in a similar way to `boolFromBit` above.
+--
+-- TODO: Implement for (System)Verilog
 unsignedFromBitVector
-  :: Backend backend
-  => Size
+  :: Size
   -> Identifier
   -> TExpr
-  -> State (BlockState backend) TExpr
-unsignedFromBitVector n = outputCoerce (BitVector n) (Unsigned n) (\i -> "unsigned(" <> i <> ")")
+  -> State (BlockState VHDLState) TExpr
+unsignedFromBitVector n =
+  outputCoerce (BitVector n) (Unsigned n) (\i -> "unsigned(" <> i <> ")")
 
 -- | Used to create an output `Bool` from a number of `Bit`s, using
 -- conjunction. Similarly to `untuple`, it returns a list of
 -- references to declared values (the inputs to the function) which
 -- should get assigned by something---usually output ports of an
 -- entity.
+--
+-- TODO: Implement for (System)Verilog
 boolFromBits
-  :: Backend backend
-  => [Identifier]
+  :: [Identifier]
   -> TExpr
-  -> State (BlockState backend) [TExpr]
+  -> State (BlockState VHDLState) [TExpr]
```

Comment:
Why was this changed to VHDL only?

Reply:
The implementation mentions `and`, which is VHDL only AFAIK. Verilog uses `&&`.